front/lib/resources/group_resource.ts

```diff
@@ -366,70 +366,199 @@ export class GroupResource extends BaseResource<GroupModel> {
+  ): Promise<Result<undefined, Error>> {
     // Checking that the user is a member of the workspace.
     const owner = auth.getNonNullableWorkspace();
-    const userResource = await UserResource.fetchById(user.sId);
-    if (!userResource) {
-      return new Err({ type: "user_not_found" });
+
+    if (users.length === 0) {
+      return new Ok(undefined);
     }
-    const workspaceMembership =
-      await MembershipResource.getActiveMembershipOfUserInWorkspace({
-        user: userResource,
-        workspace: owner,
-        transaction,
-      });
 
-    if (!workspaceMembership) {
-      return new Err({ type: "user_not_workspace_member" });
+    const userIds = users.map((u) => u.sId);
+    const userResources = await UserResource.fetchByIds(userIds);
+    if (userResources.length !== userIds.length) {
+      return new Err(
+        userIds.length === 1
+          ? new Error("The user was not found.")
+          : new Error("Some users were not found.")
+      );
+    }
+    const workspaceMemberships = await MembershipResource.getActiveMemberships({
+      users: userResources,
+      workspace: owner,
+      transaction,
+    });
+
+    if (
+      new Set(workspaceMemberships.map((m) => m.userId)).size !== userIds.length
+    ) {
+      return new Err(
+        userIds.length === 1
+          ? new Error("The user is not member of the workspace.")
+          : new Error("Some users are not members of the workspace.")
+      );
     }
 
     // Users can only be added to regular groups.
     if (this.type !== "regular") {
-      return new Err({ type: "group_not_regular" });
+      return new Err(new Error("Not a regular group, cannot be updated."));
     }
 
     // Check if the user is already a member of the group.
-    const existingMembership = await GroupMembershipModel.findOne({
-      where: {
+    const activeMembers = await this.getActiveMembers(auth);
+    const activeMembersIds = activeMembers.map((m) => m.sId);
+    const alreadyActive = userIds.filter((userId) =>
+      activeMembersIds.includes(userId)
+    );
+    if (alreadyActive.length > 0) {
+      return new Err(new Error(`User ${alreadyActive} already member.`));
+    }
+
+    // Create a new membership.
+    await GroupMembershipModel.bulkCreate(
+      users.map((user) => ({
         groupId: this.id,
         userId: user.id,
         workspaceId: owner.id,
-        startAt: { [Op.lte]: new Date() },
-        [Op.or]: [{ endAt: null }, { endAt: { [Op.gt]: new Date() } }],
-      },
+        startAt: new Date(),
+      })),
+      { transaction }
+    );
+
+    return new Ok(undefined);
+  }
+
+  async addMember(
+    auth: Authenticator,
+    user: UserType
+  ): Promise<Result<undefined, Error>> {
+    return this.addMembers(auth, [user]);
+  }
+
+  async removeMembers(
+    auth: Authenticator,
+    users: UserType[],
+    transaction?: Transaction
+  ): Promise<Result<undefined, Error>> {
+    // Checking that the user is a member of the workspace.
```

Comment:
This is not accurate, the call to `getNonNullableWorkspace` will always return a workspace if the workspace id is valid, depsite the user's permission in this workspace.